class.json-api.php

```diff
@@ -616,6 +616,85 @@ function get_avatar_url( $email, $avatar_size = null ) {
 		}
 	}
 
+	/**
+	 * Counts the number of comments on a site, excluding certain comment types.
+	 *
+	 * @param $post_id int Post ID.
+	 * @return array Array of counts, matching the output of https://developer.wordpress.org/reference/functions/get_comment_count/.
+	 */
+	public function wp_count_comments( $post_id ) {
+		global $wpdb;
+		if ( 0 !== $post_id ) {
+			return wp_count_comments( $post_id );
+		}
+
+		$counts = array(
+			'total_comments' => 0,
+			'all'            => 0,
+		);
+
+		/**
+		 * Exclude certain comment types from comment counts in the REST API.
+		 *
+		 * @since 6.9.0
+		 * @module json-api
+		 *
+		 * @param array Array of comment types to exclude (default: 'order_note', 'webhook_delivery', 'review', 'action_log')
+		 */
+		$exclude = apply_filters( 'jetpack_api_exclude_comment_types_count',
+			array( 'order_note', 'webhook_delivery', 'review', 'action_log' )
+		);
+
+		if ( empty( $exclude ) ) {
+			return wp_count_comments( $post_id );
+		}
+
+		$where = 'WHERE comment_type NOT IN (';
+		foreach( $exclude as $excluded_type ) {
+			$where .= "'" . $excluded_type . "',";
+		}
+		$where = rtrim( $where, ',' );
+		$where .= ')';
+
+
+		$count = $wpdb->get_results( "SELECT comment_approved, COUNT(*) AS num_comments
```

Comment:
Should we use `wpdb->prepare` here?

Reply:
Since we are building the WHERE clause separately (and using `esc_sql`) the prepare is not needed here -- it'll actually throw a warning wrapping it with only one parameter.